uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -688,6 +688,43 @@ private function getEntitlementCalculationMock(
     return $calculation;
   }
 
+	public function testBalanceShouldIncludeOpenLeaveRequests() {
+		$periodEntitlement = $this->createLeavePeriodEntitlementMockForBalanceTests();
+
+		$this->createLeaveBalanceChange($periodEntitlement->id, 10);
+		$this->assertEquals(10, $periodEntitlement->getBalance());
+
+		// This leave request will deduct 3 days from the entitlement
+		$this->createLeaveRequestBalanceChange(
+			$periodEntitlement->type_id,
+			$periodEntitlement->contact_id,
+			$this->leaveRequestStatuses['Approved'],
+			date('YmdHis'),
+			date('YmdHis', strtotime('+2 day'))
+		);
+
+		// This would deduct 2 days, but it's waiting approval
+		// but its expected to be included in the balance
```

Comment:
I believe you copied this from another test, right? The phrasing here ended up a bit odd. I think it's enough to just say how many days will be deducted